### ISSUE TITLE
undefined response

### ISSUE THREAD

**Issue by CharlesCollas** [bojand/mailgun-js#189]
Issue happened today, didn't change anything to my code, was working great before (didn't upgrade my mailgun-js package either).

mailgun-js 2.5.1
node v0.12.0

It works when I use curl to send an email.

Sample source code:

```
var api_key = 'key-a3xxxxxxxxxx';
var domain = 'mg.domain.com';
var mailgun = require('mailgun-js')({apiKey: api_key, domain: domain});

var data = {
    from: 'Me <mailgun@mg.domain.com>',
    to: 'test@example.org',
    subject: 'Hi',
    text: 'Hello',
    'h:Reply-To': 'myself@example.org'
};

mailgun.messages().send(data, function (error, body) {
    console.log(body);
});
```

the console.log(body); returns undefined.

I saw a previous issue closed about it but for me it worked, it's not about a URL domain VS only the domain, I've always used only the domain (ex: "mg.domain.com", which is active).

Thanks!

**Comment by bojand** [comment]
Hello, what previous issue are you referring to? There is no version 2.5.1 of this module. Also Node v0.12.0 is [End of Life](https://github.com/nodejs/Release) and you should probably update. This module requires Node 6 or above.

**Comment by CharlesCollas** [comment]
Thanks for your answer bojand,

I'm referring to this issue: https://github.com/bojand/mailgun-js/issues/146

You're right, I was just afraid to break everything in my code by updating Node ...

But this module worked well for me even with Node v0.12.0 until today!

Do you know how I could fix it?

**Comment by bojand** [comment]
Really don't know... What version of the module are you using as there is no version 2.5.1...? Are you sure you're using the proper domain for your account? Is this a test domain? The latest version is working fine from some tests I just performed.

**Comment by CharlesCollas** [comment]
Nevermind the version of the module I'm using is 0.7.7

Yes proper domain, not a test domain.

Maybe I should upgrade this module to last version?

**Comment by tluck234** [comment]
Mailgun updated their certs and caused an issue for myself. It could be related as we have similar node configs.

**Comment by CharlesCollas** [comment]
Yes thanks @tluck234 , I've had the same reply from them, it is related.

Have you managed to fix this?

Restarting my app didn't work out, it seems like I need to clear the certificate cache but I can't manage to do so.

**Comment by tluck234** [comment]
I have not been able to correct it yet despite digging into certificate settings, etc. The only resolution at the moment for myself was to set the mailgun-js init to HTTP such as:

{apiKey: api_key, domain: domain, "protocol":"http:", "port":80});

**Comment by samotage** [comment]
Seems the app I am maintaining is way old...  My fix similar to above is a hack of the node package:

nodemailer-mailgunapi-transport

To change in index.js
//var baseUrl = "https://api.mailgun.net/v2";
var baseUrl = "http://api.mailgun.net:80/v2";

Seems to work.  Ugly - however mailgun support seems to be a black hole...

**Comment by tluck234** [comment]
I did notice some irregular behavior after making the change I mentioned above. Not 100% sure if it's just general delay for deliveries or not, but we had a number of clients mention they were not getting **some** of the emails. Our logs seemed to suggest it as well.

With that said, I forked the repo and changed the request.js file to make SSL requests using rejectUnathorized: false as a bandaid. Our email service seems to be running as it was before the mailgun certificate change now.
